docs/application/native/guides/text-input/tts-engine.md

```diff
@@ -0,0 +1,369 @@
+        {
+            ttse_request_callback_s callback = {
+                /* Add the mandatory callbacks */
+            };
+
+            ttse_main(argc, argv, &callback);
+
+            ttse_set_private_data_set_cb(private_data_set_cb);
+            ttse_set_private_data_requested_cb(private_data_requested_cb);
+            ttse_set_activated_mode_changed_cb(activated_mode_changed_cb);
+
+            ...
+        }
+        ```
+
+4. Define the optional callbacks:
+    ```
+    static int
+    private_data_set_cb(const char* key, const char* data)
+    {
+        return 0;	/* private data save succeed */
+    }
+
+    static int
+    private_data_requested_cb(const char* key, char** data)
+    {
+        /* found key */
+        *data = strdup("data");
+        return 0;	/* private data set succeed */
+    }
+
+    static void
+    activated_mode_changed_cb(int activated_mode)
+    {
+        return ;	/* get activated modes changed succeed */
+    }
+    ```
+
+
+<a name="parameters"></a>
+## TTSE Parameters
+
+You can set the following parameters about the TTSE:
+
+- Send the current state using the `ttse_send_error()` function and send the error to a engine service user.
+
+    ```
+    void
+    send_error(void)
+    {
+        int ret = TTSE_ERROR_NONE;
+
+        ret = ttse_send_error(TTSE_ERROR_NONE, "msg", NULL);
+        if(TTSE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+
+<a name="info_retrieval"></a>
+## TTSE Information Retrieval
+
+You can get the following information about the TTSE:
+
+- Get the speed range using the `ttse_get_speed_range()` function.
+
+    ```
+    void
+    get_speed_range(void)
+    {
+        int ret = TTSE_ERROR_NONE;
+        int min_speed = 0;
+        int normal_speed = 0;
+        int max_speed = 0;
+
+        ret = ttse_get_speed_range(&min_speed, &normal_speed, &max_speed);
+        if(TTSE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+- Get the pitch range using the `ttse_get_pitch_range()` function.
+
+    ```
+    void
+    get_pitch_range(void)
+    {
+        int ret = TTSE_ERROR_NONE;
+        int min_pitch = 0;
+        int normal_pitch = 0;
+        int max_pitch = 0;
+
+        ret = ttse_get_pitch_range(&min_pitch, &normal_pitch, &max_pitch);
+        if(TTSE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+- Get the activated modes using the `ttse_get_activated_mode()` function. According to modes of TTS clients connected to TTS engine service, the output argument `activated_mode` of `ttse_get_activated_mode()` is a bit sequence as `#TTSE_MODE_MASK_DEFAULT | #TTSE_MODE_MASK_SCREEN_READER`.
+
+    ```
+    void
+    get_activated_modes(void)
+    {
+        int ret = TTSE_ERROR_NONE;
+        int activated_mode;
+
+        ret = ttse_get_activated_mode(&activated_mode);
+        if(TTSE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+
+<a name="send_result"></a>
+## TTSE Send results
```

Comment:
```suggestion
## TTSE send results
```

Reply:
I changed this subject like the following:
```suggestion
## Send results
```

Please review this again. Thank you.